src/utils/file/index.ts

```diff
@@ -188,6 +217,54 @@ export async function convertForPreview(file: File, fileBlob: Blob) {
     return fileBlob;
 }
 
+export function fileIsArchived(file: File) {
+    if (
+        !file ||
+        !file.magicMetadata ||
+        !file.magicMetadata.data ||
+        typeof file.magicMetadata.data === 'string' ||
+        typeof file.magicMetadata.data.visibility === 'undefined'
+    ) {
+        return false;
+    }
+    return file.magicMetadata.data.visibility === VISIBILITY_STATE.ARCHIVED;
+}
+
+export async function changeFilesVisibility(
+    files: File[],
+    selected: SelectedState,
+    visibility: VISIBILITY_STATE
+) {
+    const worker = await new CryptoWorker();
+    const selectedFiles = getSelectedFiles(selected, files);
+    const updatedFiles: File[] = [];
+    for (const file of selectedFiles) {
+        if (!file.magicMetadata) {
+            file.magicMetadata = NEW_MAGIC_METADATA;
+        }
+        if (typeof file.magicMetadata.data === 'string') {
+            logError(Error('magic metadata not decrypted'), '');
+            return;
+        }
+        const updatedMagicMetadataProps: MagicMetadataProps = {
+            ...file.magicMetadata.data,
+            visibility,
+        };
```

Comment:
For future, shall we add comment here [1] about the expected behaviour here during merge? There seems to be some JS specific voodoo/ syntax here, which a non-js/newbie programmer might not be able to understand at first go :) 

[1] https://github.com/ente-io/frame/blob/2bbf49384fe3b289180e011b901e3928664d9581/lib/services/file_magic_service.dart#L49